### PR TITLE
Use freezegun in tests and add last_fetched to snapshots

### PR DESCRIPTION
**Pull request by ludeeus**
SSIA

### CODE REVIEW

File: tests/patch_time.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from homeassistant import util
+from homeassistant.util import dt as dt_util
+
+
+class CustomFakeDatetime(freezegun.api.FakeDatetime):  # type: ignore[name-defined]
+    """Modified to workaround tz problem."""
+
+    @classmethod
+    def _tz_offset(cls):
+        return datetime.timedelta(hours=0)
```
Comment (emontnemery) [inline tests/patch_time.py:16]: Can we better explain what this does? What timezone problem is fixed?

File: tests/patch_time.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+def _utcnow() -> datetime.datetime:
+    """Make utcnow patchable by freezegun."""
+    return datetime.datetime.now(tz=datetime.UTC)
+
+
+dt_util.utcnow = _utcnow  # type: ignore[assignment]
+util.utcnow = _utcnow  # type: ignore[assignment]
```
Comment (emontnemery) [inline tests/patch_time.py:25]: Why is this needed?

Comment (ludeeus) [reply tests/patch_time.py:25]: These are needed for the WSClient, without it auth will fail.

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -35,6 +35,11 @@ jobs:
           scripts/install/core
           scripts/install/frontend
 
+      - name: ⏲️ Set time zone
+        uses: MathRobin/timezone-action@v1.1
+        with:
+          timezoneLinux: 'Asia/Singapore'
```
Comment (ludeeus) [inline .github/workflows/pytest.yml:41]:
```suggestion
        uses: szenius/set-timezone@v2.0
        with:
          timezoneLinux: 'Europe/Amsterdam'
```

File: .github/workflows/pytest.yml [hunk 4]
```diff
@@ -69,6 +74,11 @@ jobs:
           scripts/install/core_dev
           scripts/install/frontend
 
+      - name: ⏲️ Set time zone
+        uses: MathRobin/timezone-action@v1.1
+        with:
+          timezoneLinux: 'Asia/Singapore'
```
Comment (ludeeus) [inline .github/workflows/pytest.yml:80]:
```suggestion
        uses: szenius/set-timezone@v2.0
        with:
          timezoneLinux: 'Europe/Amsterdam'
```

**Review by emontnemery — APPROVED** [review]
Nice!

**Comment by emontnemery** [comment]
Tests were failing with the timezone action because https://github.com/hacs/integration/pull/3570 is not merged
